app/src/main/java/com/android/universe/ui/emailVerification/EmailVerificationScreen.kt

```
@@ -0,0 +1,277 @@
+import androidx.compose.ui.unit.dp
+import com.android.universe.ui.navigation.NavigationTestTags
+import com.android.universe.ui.profileCreation.AddProfileScreenTestTags
+import com.android.universe.ui.theme.Dimensions
+import com.google.firebase.Firebase
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.auth
+
+object EmailVerificationScreenTestTags {
+  const val ICON_BOX = "ICON_BOX"
+  const val HEADLINE_TEXT = "HEADLINE_TEXT"
+  const val MESSAGE_TEXT = "MESSAGE_TEXT"
+  const val INSTRUCTIONS_TEXT = "INSTRUCTIONS_TEXT"
+  const val COUNTDOWN_TEXT = "COUNTDOWN_TEXT"
+  const val RESEND_BUTTON = "RESEND_BUTTON"
+}
+
+/**
+ * A composable screen that handles the email verification process.
+ *
+ * This screen is displayed after a user registers with an email and password. It manages the UI
+ * state for sending the verification email, waiting for the user to verify, handling failures in
+ * sending the email, and providing an option to resend the verification link.
+ *
+ * It observes the user's verification status from Firebase and automatically navigates upon
+ * successful verification.
+ *
+ * @param user The current `FirebaseUser` whose email needs to be verified. Defaults to the
+ *   currently signed-in user.
+ * @param onSuccess A lambda function to be invoked when the email has been successfully verified.
+ *   Typically used for navigation to the next screen (e.g., home or profile creation).
+ * @param onBack A lambda function to be invoked when the user clicks the back button in the top app
+ *   bar. Typically used for navigating back to the previous screen (e.g., login or registration).
+ * @param viewModelInstance An unitialized [EmailVerificationViewModel] that manages the state and
+ *   logic for this screen. This is provided for testability and defaults to a new instance.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EmailVerificationScreen(
+    user: FirebaseUser = Firebase.auth.currentUser!!,
```

If the user is null, the app will crash. I would prefer to throw an error that doing that.